esteid/mobileid/base.py

```diff
@@ -0,0 +1,423 @@
+        return AuthenticateResult(
+            session_id=result["sessionID"],
+            digest=hash_value,
+            hash_type=hash_type,
+            verification_code=get_verification_code(hash_value),
+        )
+
+    def status(self, session_id, digest, timeout=10000):
+        """
+        Retrieve auth session result from Mobile-ID backend
+
+        see https://github.com/SK-EID/MID#33-status-of-signing-and-authentication
+
+        :param session_id: session ID, from auth result
+        :param digest: hash value that was used to initiate auth
+        :param int timeout: Request long poll timeout value in milliseconds (Note: server uses a default
+                         if client does not send it)
+        :rtype: AuthenticateStatusResult
+        """
+        data = self._get_session_response(self.Actions.AUTH, session_id, timeout)
+
+        # signature.value (str): Signature value, base64 encoded.
+        signature_value = base64.b64decode(data["signature"]["value"])
+
+        # signature.algorithm (str): Signature algorithm, in the form of sha256WithRSAEncryption
+        signature_algorithm = data["signature"]["algorithm"]
+        assert signature_algorithm[:6].upper() in HASH_ALGORITHMS
+
+        # cert: Certificate value, DER+Base64 encoded
+        cert_value = base64.b64decode(data["cert"])
+
+        try:
+            pyasice.verify(cert_value, signature_value, digest, signature_algorithm[:6], prehashed=True)
+        except pyasice.SignatureVerificationError:
+            raise SignatureVerificationError(self.msg("signature_mismatch"))
+
+        return AuthenticateStatusResult(
+            certificate=cert_value,
+            signature=signature_value,
+            signature_algorithm=signature_algorithm,
+        )
+
+    def sign(
+        self,
+        id_code,
+        phone_number,
+        signed_data,
+        message=None,
+        language=None,
+        hash_type=HASH_SHA256,
+    ):
+        """Initiate a signature session.
+
+        Differences from authentication session:
+        - endpoint
+        - using signed data instead of a random string
+
+        https://github.com/SK-EID/MID#32-initiating-signing-and-authentication
+
+        :param str id_code: National identity number
+        :param str phone_number: Client's phone number
+        :param bytes signed_data: Binary data to sign
+        :param str message: Text to display for authentication consent dialog on the mobile device
+        :param str language: choices: ENG, EST, LIT, RUS. Defaults to ENG
+        :param str hash_type: Hash algorithm used to sign data
+        :return SignResult: Result of the request
+        """
+        assert id_code
+        assert phone_number
+
+        hash_type = hash_type.upper()
+        assert hash_type in HASH_ALGORITHMS
+
+        message = str(message or self.msg("display_text"))
+        assert len(message) <= 20, f"Display text can not exceed 20 chars: got '{message}'"
```

Comment:
This requirement should be added to the parameter doc string, I can see cases where might become a surprise exception.

Reply:
Thanks, done!